src/d/d_wood.cpp

```diff
@@ -4,184 +4,917 @@
+enum UnitFlags {
+  Unit_IsActive = 1 << 0,
+  Unit_IsFrustumCulled = 1 << 1,
+  Unit_IsChopped = 1 << 2,
+};
+
+//-----------------------------------------
+// Globals
+//-----------------------------------------
+const s32 L_Room_Max = 64;
+u8 L_Alpha_Cutoff = 0x80;
+const GXColor l_shadowColor = {0x00, 0x00, 0x00, 0x64};
+u8 l_matDL[8 /* TODO: real length */] = {};
+
+u32 l_Oba_swood_b_cutDL_SIZE = 0xc0;
+u32 l_Oba_swood_bDL_SIZE = 0x100;
+u8 l_Oba_swood_b_cutDL[0xc0];
+u8 l_Oba_swood_bDL[0x100];
+
+const float l_Ground_check_y_offset = 100.0f;
+const float l_Ground_check_unk0 = 1.0E+9;
+const double l_Ground_check_unk1 = 0.5;
+const double l_Ground_check_unk2 = 3.0;
+float kGroundHeightBias = 1.0f;
+
+struct AnimAttrs {
+  /* 0x0 */ s16 unkUShort0;
+  /* 0x2 */ s16 unkShort1;
+  /* 0x4 */ s16 unkShort2;
+  /* 0x6 */ s16 unkShort3;
+  /* 0x8 */ float unkFloat;
+};
+
+STATIC_ASSERT(sizeof(AnimAttrs) == 0xc);
+
+// Constants
+struct Attr_c {
+  /* 0x0 */ AnimAttrs base[4][2];
+  /* 0x60 */ u8 kCutCooldown;           // = 23
+  /* 0x61 */ u8 kCutPosOffsetX;         // = 0
+  /* 0x64 */ float kCutInitVelY;        // = 18.0f
+  /* 0x68 */ float kCutYAccel;          // = -3.0f (Units per frame per frame)
+  /* 0x6C */ float kCutZVel;            // = 2.5f (Units per frame)
+  /* 0x70 */ float kCutPitchVel;        // = 8 (shortRad per frame)
+  /* 0x74 */ float kUncutShadowScale;   // = 1.5f
+  /* 0x78 */ float kCutShadowScale;     // = 0.3f
+  /* 0x7c */ float kCollisionRad;       // = 80.0f;
+  /* 0x80 */ float kCollisionRadCut;    // = 15.0f;
+  /* 0x84 */ float kCollisionHeight;    // = 80.0f;
+  /* 0x88 */ float kCollisionHeightCut; // = 15.0f;
+  /* 0x8c */ float kClipRadius;         // = 100.0f;
+  /* 0x90 */ float kClipCenterYOffset;  // = 40.0f;
+  /* 0x94 */ u8 kPushBackCountdown;     // = 23
+};
+
+static const Attr_c L_attr = {};
+
+//-----------------------------------------
+// Helpers
+//-----------------------------------------
+
+// This is a 3-step sqrt estimate, but it also contains an extra multiply by the
+// supplied magnitude. I am not 100% sure what it is accomplishing.
+static inline double inv_sqrt(float mag) {
+  if (mag > 0.0f) {
+    double root = __frsqrte(mag);
+    double res = mag;
+    res = 0.5 * root * (3.0 - res * (root * root)) * res * mag;
+    res = 0.5 * root * (3.0 - res * (root * root)) * res * mag;
+    res = 0.5 * root * (3.0 - res * (root * root)) * res;
+    return res;
+  } else {
+    return mag;
+  }
+}
+
+//-----------------------------------------
+// Classes
+//-----------------------------------------
 
 /* 800BD678-800BD710       .text __ct__Q25dWood5Anm_cFv */
 dWood::Anm_c::Anm_c() {
-    /* Nonmatching */
+  int iVar1;
+  int iVar2;
+
+  PSMTXIdentity(mModelMtx);
+  PSMTXIdentity(mTrunkModelMtx);
+  mMode = Mode_Max;
+  this->mCountdown = 0;
+  this->mWindDir = 0;
+  this->mWindPow = 0.0;
+  this->mPosOffsetY = 0.0;
+  this->mPosOffsetZ = 0.0;
+  this->mVelY = 0.0;
```

Comment:
Remove `this->`

Reply:
Is there a way to have Ghidra not add "this->" in the decompile window?

Reply:
I doubt it, ghidra is effectively showing C++ as if it was pseudo-C code, and typing `this->` is necessary is C.